dd-sdk-android/src/main/kotlin/com/datadog/android/Datadog.kt

```diff
@@ -278,8 +278,14 @@ object Datadog {
         // so some things may yet not be initialized -> not accessible, some things may already be
         // initialized and be not mutable anymore
         additionalConfiguration[DD_SOURCE_TAG]?.let {
-            if (it.toString().isNotBlank()) {
-                CoreFeature.sourceName = it.toString()
+            if (it is String && it.isNotBlank()) {
```

Comment:
What if it isn't a String ? Shouldn't we still deal with it ?

Reply:
What can be the case for that and what type can be instead? I think all external platforms have string as primitive.